x/gamm/types/msgs_test.go

```diff
@@ -7,13 +7,273 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/stretchr/testify/require"
 
-	"github.com/osmosis-labs/osmosis/v13/x/gamm/pool-models/stableswap"
 	gammtypes "github.com/osmosis-labs/osmosis/v13/x/gamm/types"
+	swaproutertypes "github.com/osmosis-labs/osmosis/v13/x/swaprouter/types"
 
 	"github.com/osmosis-labs/osmosis/v13/app/apptesting"
 	appParams "github.com/osmosis-labs/osmosis/v13/app/params"
 )
 
+func TestMsgSwapExactAmountIn(t *testing.T) {
+	appParams.SetAddressPrefixes()
+	pk1 := ed25519.GenPrivKey().PubKey()
+	addr1 := sdk.AccAddress(pk1.Address()).String()
+	invalidAddr := sdk.AccAddress("invalid")
+
+	createMsg := func(after func(msg gammtypes.MsgSwapExactAmountIn) gammtypes.MsgSwapExactAmountIn) gammtypes.MsgSwapExactAmountIn {
+		properMsg := gammtypes.MsgSwapExactAmountIn{
+			Sender: addr1,
+			Routes: []swaproutertypes.SwapAmountInRoute{{
+				PoolId:        0,
+				TokenOutDenom: "test",
+			}, {
+				PoolId:        1,
+				TokenOutDenom: "test2",
+			}},
+			TokenIn:           sdk.NewCoin("test", sdk.NewInt(100)),
+			TokenOutMinAmount: sdk.NewInt(200),
+		}
```

Comment:
I'm a little confused on this as well, I thought we returned a panic when calling a swapExactAmountIn msg from gamm side?

Reply:
These tests are being re-added. See main: https://github.com/osmosis-labs/osmosis/blob/main/x/gamm/types/msgs_test.go

I'm not adding anything new, only bringing back previously removed messages and tests